cpu/samd5x/cpu.c

```diff
@@ -41,21 +40,29 @@ static void dfll_init(void)
 
     OSCCTRL->DFLLCTRLB.reg = reg;
     OSCCTRL->DFLLCTRLA.reg = OSCCTRL_DFLLCTRLA_ENABLE;
+
+    while (!OSCCTRL->STATUS.bit.DFLLRDY) {}
 }
-#endif
 
 static void fdpll0_init(uint32_t f_cpu)
 {
+    /* We source the DPLL from 32kHz GCLK1 */
     const uint32_t LDR = ((f_cpu << 5) / 32768);
 
+    /* disable the DPLL before changing the configuration */
+    OSCCTRL->Dpll[0].DPLLCTRLA.bit.ENABLE = 0;
+    while (OSCCTRL->Dpll[0].DPLLSYNCBUSY.reg) {}
+
+    /* set DPLL clock source */
     GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0].reg = GCLK_PCHCTRL_GEN(1) | GCLK_PCHCTRL_CHEN;
+    while (!(GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0].reg & GCLK_PCHCTRL_CHEN)) {}
 
     OSCCTRL->Dpll[0].DPLLRATIO.reg = OSCCTRL_DPLLRATIO_LDRFRAC(LDR & 0x1F)
-                       | OSCCTRL_DPLLRATIO_LDR((LDR >> 5) - 1);
+                                   | OSCCTRL_DPLLRATIO_LDR((LDR >> 5) - 1);
 
+    /* Without LBYPASS, startup takes very long, reson unknown. */
```

Comment:
Reason is sillicon bug,
check errata [there](http://ww1.microchip.com/downloads/en/DeviceDoc/SAM-D5X-E5X-Family-Silicon-Errata-and-Data-Sheet-Clarification-80000748F.pdf). section 2.13

Reply:
Ah thank you! I should read the errata sheet more often